File: bot/src/main/java/com/lm_pakkanen/radio_pele_java/commands/BaseCommand.java
package com.lm_pakkanen.radio_pele_java.commands;

import com.lm_pakkanen.radio_pele_java.models.exceptions.InvalidChannelException;
import net.dv8tion.jda.api.entities.channel.ChannelType;
import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;
import net.dv8tion.jda.api.entities.channel.unions.MessageChannelUnion;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;

public abstract class BaseCommand extends ListenerAdapter {

  /**
   * @param event which initiated the command.
   * @return the text channel where the command was initiated.
   * @throws InvalidChannelException
   */
  public TextChannel getTextChan(SlashCommandInteractionEvent event)
      throws InvalidChannelException {

    final MessageChannelUnion messageChan = event.getChannel();

    if (!messageChan.getType().equals(ChannelType.TEXT)) {
      throw new InvalidChannelException();
    }

    return messageChan.asTextChannel();
  }
}

File: bot/src/main/kotlin/com/lm_pakkanen/radio_pele_java/commands/BaseCommand.kt
package com.lm_pakkanen.radio_pele_java.commands

import com.lm_pakkanen.radio_pele_java.models.exceptions.InvalidChannelException
import net.dv8tion.jda.api.entities.channel.ChannelType
import net.dv8tion.jda.api.entities.channel.concrete.TextChannel
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter

abstract class BaseCommand : ListenerAdapter() {
    /**
     * @param event which initiated the command.
     * @return the text channel where the command was initiated.
     * @throws InvalidChannelException
     */
    @Throws(InvalidChannelException::class)
    fun getTextChan(event: SlashCommandInteractionEvent): TextChannel {
        val messageChan = event.channel

        if (messageChan.type != ChannelType.TEXT) {
            throw InvalidChannelException()
        }

        return messageChan.asTextChannel()
  }
}
